akka-projection-grpc/src/test/scala/akka/projection/grpc/internal/ProtoAnySerializationSpec.scala

```diff
@@ -59,48 +78,97 @@ class ProtoAnySerializationSpec extends ScalaTestWithActorTestKit with AnyWordSp
         .setTypeUrl(typeUrl)
         .setValue(ByteString.copyFrom(value, StandardCharsets.UTF_8))
         .build()
-      val pbAny = protoAnySerialization.encode(event)
+      val pbAny = serializationJava.serialize(event)
       pbAny.typeUrl shouldBe typeUrl
       val deserializedEvent =
-        protoAnySerialization.decode(pbAny).asInstanceOf[PbAny]
+        serializationJava.deserialize(pbAny).asInstanceOf[PbAny]
       deserializedEvent.getTypeUrl shouldBe typeUrl
       deserializedEvent.getValue.toString(StandardCharsets.UTF_8) shouldBe value
     }
 
-    "pass through ScalaPb Any and decode it as Java proto Any" in {
+    "pass through ScalaPb Any and decode it as preferred Any Any" in {
       val value = "hello"
       val typeUrl = "type.my.io/custom"
       val event =
         ScalaPbAny(typeUrl, ByteString.copyFrom(value, StandardCharsets.UTF_8))
-      val pbAny = protoAnySerialization.encode(event)
+      val pbAny = serializationScala.serialize(event)
+      pbAny.typeUrl shouldBe typeUrl
+
+      val deserializedEventScala =
+        serializationScala.deserialize(pbAny).asInstanceOf[ScalaPbAny]
+      deserializedEventScala.typeUrl shouldBe typeUrl
+      deserializedEventScala.value.toString(StandardCharsets.UTF_8) shouldBe value
+
+      val deserializedEventJava =
+        serializationJava.deserialize(pbAny).asInstanceOf[PbAny]
+      deserializedEventJava.getTypeUrl shouldBe typeUrl
+      deserializedEventJava.getValue.toString(StandardCharsets.UTF_8) shouldBe value
+    }
+
+    "pass through Java proto Any with Google typeUrl" in {
+      val instant = Instant.now()
+      val value =
+        com.google.protobuf.Timestamp
+          .newBuilder()
+          .setSeconds(instant.getEpochSecond)
+          .setNanos(17)
+          .build()
+      val typeUrl = "type.googleapis.com/google.protobuf.Timestamp"
+      val event = PbAny
+        .newBuilder()
+        .setTypeUrl(typeUrl)
+        .setValue(value.toByteString)
+        .build()
+      val pbAny = serializationJava.serialize(event)
+      pbAny.typeUrl shouldBe "type.googleapis.com/google.protobuf.Timestamp"
+      val deserializedEvent = serializationJava.deserialize(pbAny)
+      deserializedEvent.getClass shouldBe classOf[com.google.protobuf.Timestamp]
+      deserializedEvent shouldBe event
```

Comment:
This test is failing, and rather confusing that it's trying to deserialize instead of just pass through "same" Any